packages/sdk/src/types/asset.ts

```diff
@@ -0,0 +1,34 @@
+import type { Address } from 'viem'
+
+import type { SupportedChainId } from '@/constants/supportedChains.js'
+
+/**
+ * Asset type inspired by EIP 7811 but adapted for multi-chain asset definitions
+ */
+export interface Asset {
+  /** Multi-chain address mapping - use 'native' for native tokens like ETH */
+  address: Partial<Record<SupportedChainId, Address | 'native'>>
```

Comment:
question - why can the value be of type `Address | 'native'`? that seems like it could get messy because everywhere we use an address we'll have to check if it's native. typically native ether is just set to the `0x0000000000000000000000000000000000000000` address
<img width="711" height="796" alt="Screenshot 2025-09-17 at 5 13 37 PM" src="https://github.com/user-attachments/assets/53404097-8c9a-43e3-b361-2d117e369157" />

Reply:
good find, i had it as an option for a moment, but removed it